### PR TITLE
Tail sweep attack for alien queen

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Allows the alien queen to spin around with a roar, throwing back/stunning everyone within two tiles.

EDIT: Wow what a waste of a get I hope you're happy @Wjohn xenos win

### CODE REVIEW

**Comment by GunHog** [comment]
This would be AMAZING if you could cut off the tail of a Xeno queen and stick it on your butt for tail sweeping action! ~~and ERP!~~

**Comment by Aranclanos** [comment]
Remove all the copypaste, including strings of text and spinning.
Remove all the new useless vars that are redundant because of the new typepath.
Post a gif or webm of the effect itself.

**Comment by WJohn** [comment]
:OOOOO

File: code/datums/spells/wizard.dm [hunk 1]
```diff
@@ -306,30 +310,64 @@
 	var/list/thrownatoms = list()
 	var/atom/throwtarget
 	var/distfromcaster
-	playsound(user, "sound/magic/Repulse.ogg", 50, 1, -1)
+	if(xenomorph)
```
Comment (optimumtact) [inline code/datums/spells/wizard.dm]: I'd prefer you'd subclass the spell class rather than add a var to it

File: code/datums/spells/wizard.dm [hunk 2]
```diff
@@ -297,7 +297,11 @@
 	range = 5
 	cooldown_min = 150
 	selection_type = "view"
+	var/xenomorph = 0
```
Comment (Iamgoofball) [inline code/datums/spells/wizard.dm]: aaaaaa

Comment (duncathan) [inline code/datums/spells/wizard.dm]: what
what does this mean even
why do all spells have this var???????

File: code/datums/spells/wizard.dm (hunk 1 above)
Comment (duncathan) [inline code/datums/spells/wizard.dm]: OH

File: code/datums/spells/wizard.dm [hunk 3]
```diff
@@ -306,30 +310,64 @@
...
 	for(var/turf/T in targets) //Done this way so things don't get thrown all around hilariously.
 		for(var/atom/movable/AM in T)
 			thrownatoms += AM
 
 	for(var/atom/movable/AM in thrownatoms)
 		if(AM == user || AM.anchored) continue
-
 		var/obj/effect/overlay/targeteffect	= new /obj/effect/overlay{icon='icons/effects/effects.dmi'; icon_state="shieldsparkles"; mouse_opacity=0; density = 0}()
-		AM.overlays += targeteffect
+		if(xenomorph)
```
Comment (duncathan) [inline code/datums/spells/wizard.dm]: OH GOD NO

File: code/datums/spells/wizard.dm [hunk 4]
```diff
@@ -306,30 +310,64 @@
...
+				sleep(1)
+				user.dir = 2
+		else
+			AM.overlays += targeteffect
 		throwtarget = get_edge_target_turf(user, get_dir(user, get_step_away(AM, user)))
 		distfromcaster = get_dist(user, AM)
 		spawn(10)
-			AM.overlays -= targeteffect
-			qdel(targeteffect)
+			if(!xenomorph)
```
Comment (duncathan) [inline code/datums/spells/wizard.dm]: OH SWEET JESUS HELP ME

Comment (duncathan) [inline code/datums/spells/wizard.dm]: pls make subtype + procs thank you

**Comment by WJohn** [comment]
Maybe ask @cuboos for a tail whip sound?

**Comment by Cuboos** [comment]
What'cha need?

**Comment by KorPhaeron** [comment]
I'll re-open it when I have time to redo it then

**Comment by Xhuis** [comment]
#12000

**Comment by Aranclanos** [comment]
looks ok

**Comment by Cuboos** [comment]
So... what exactly are you looking for?

**Comment by WJohn** [comment]
Some sort of loud whip sound. Maybe even with a whipcrack. A low pitch would be nice, rather than the high pitch you get from a belt whip.

**Comment by Cuboos** [comment]
Alright when i get some time today i'll see what i can come up with, if i could see the animation i'd have an easier time coming up with a sound for it.

**Comment by WJohn** [comment]
Well the animation is just the queen making a spin, but maybe you could look towards the finale of aliens when ripley's fighting the queen and there's lots of tail whipping going on, that might be a good base for the sound.

**Comment by Cuboos** [comment]
https://dl.dropboxusercontent.com/u/9654207/Alium_swing_tail.ogg
https://dl.dropboxusercontent.com/u/9654207/Alium_Whip_human.ogg
https://dl.dropboxusercontent.com/u/9654207/Alium_Whip_borg.ogg

How's This?

**Comment by Anonus** [comment]
This should show up in the Alien tab and not the Spells tab because it's only TECHNICALLY a spell.

**Comment by bgobandit** [comment]
alium dominatrix erp

**Comment by Cuboos** [comment]
Bandit, you always know how to excite me.

**Comment by WJohn** [comment]
I don't know what it is about the first one but it sort of hurts my ears right at the start. I think it's how sudden it goes from void to whip sound? I don't know.

The other two are great.

**Comment by Razharas** [comment]
@KorPhaeron

**Comment by KorPhaeron** [comment]
I never did add the sound or test the spin speed. Oh well

**Comment by Razharas** [comment]
I did add the sound and the tested the spin speed
It looks perfect

**Comment by Razharas** [comment]
I also added tailspin to praetorians because why not

**Comment by KorPhaeron** [comment]
The manual merge power has gone to your head you're outta control

**Comment by WJohn** [comment]
O-ok.

**Pull request merged**